src/sentry/static/sentry/app/components/idBadge/userBadge.tsx

```diff
@@ -89,16 +111,21 @@ const StyledEmail = styled('div')`
   ${overflowEllipsis};
 `;
 
-const StyledName = styled(({useLink, to, ...props}) => {
+type NameProps = {
+  useLink: boolean;
+  hideEmail: boolean;
+} & Link['props'];
+
+const StyledName = styled(({useLink, to, ...props}: NameProps) => {
   const forwardProps = omit(props, 'hideEmail');
   return useLink ? <Link to={to} {...forwardProps} /> : <span {...forwardProps} />;
 })`
-  font-weight: ${p => (p.hideEmail ? 'inherit' : 'bold')};
+  font-weight: ${(p: NameProps) => (p.hideEmail ? 'inherit' : 'bold')};
```

Comment:
Can we use the generics form with `styled()`? I believe it'll be required with emotion 10's type checking. Using generics should make this redundant.